pyMongo_task.py

```diff
@@ -0,0 +1,69 @@
+import datetime
+from pymongo import MongoClient
+import sys, csv, sqlite3
+
+
+mongo = MongoClient(host='localhost', port=27017)
+
+def _connect_mongo(host, port, username, password, db):
+
+    if username and password:
+        mongo_uri = 'mongodb://%s:%s@%s:%s/%s' % (username, password, host, port, db)
+        conn = MongoClient(mongo_uri)
+    else:
+        conn = MongoClient(host, port)
+
+    return conn[db]
+
+#csv files to db
+csv_projects = open('D://Python_Learn//csv_projects.csv', 'r') #file with projects
+csv_tasks = open('D://Python_Learn//csv_tasks.csv', 'r') #file with tasks
+
+reader_proj = csv.DictReader( csv_projects )
+reader_tasks = csv.DictReader( csv_tasks )
+db = mongo.itest_db #access to my db not created yet
+db.projects.drop() # Removes a projects collection
+db.tasks.drop() # Removes a tasks collection
+
+header_projects = ["name", "description", "deadline"]
+#insert projects to projects_table
+for each in reader_proj:
+    row={}
+    for field in header_projects:
+        row[field]=each[field]
+    db.projects.insert_one(row)
+
+header_tasks = ["id", "priority", "details", "status", "deadline", "completed", "project"]
+#insert_tasks_to tasks table
+for each in reader_tasks:
+    row={}
+    for val in header_tasks:
+        row[val]=each[val]
+    db.tasks.insert_one(row)

```

Comment:
All this code can be simplified to:
```
db.projects.insert_many(reader_proj)
db.tasks.insert_many(reader_tasks)
```
Using insert_one is normal when we have a small input file.

A batch of documents can be inserted by passing a list to the **insert_many()** method. PyMongo will automatically split the batch into smaller sub-batches based on the maximum message size accepted by MongoDB, supporting very large bulk insert operations.

Executing write operations in batches reduces the number of network round trips, increasing write throughput.